packages/flutter/lib/src/painting/debug_overflow_indicator.dart

```diff
@@ -0,0 +1,392 @@
+      if (textDirection != null) {
+        switch (textDirection) {
+          case TextDirection.ltr:
+            label = 'END OVERFLOWED BY $pixels PIXELS';
+            break;
+          case TextDirection.rtl:
+            label = 'BEGIN OVERFLOWED BY $pixels PIXELS';
+            break;
+        }
+      } else {
+        label = 'RIGHT OVERFLOWED BY $pixels PIXELS';
+      }
+      regions.add(new _OverflowRegionData(
+        rect: markerRect,
+        label: label,
+        labelOffset: markerRect.centerRight -
+            const Offset(_debugMarkerFontSize + _debugMarkerLabelPadding, 0.0),
+        rotation: -math.PI / 2.0,
+      ));
+    }
+    if (overflow.top > 0.0) {
+      final String pixels = _formatPixels(overflow.top);
+      final Rect markerRect = new Rect.fromLTWH(
+        0.0,
+        0.0,
+        _overflowContainerRect.width,
+        _overflowContainerRect.height * _kMarkerSize,
+      );
+      regions.add(new _OverflowRegionData(
+        rect: markerRect,
+        label: 'TOP OVERFLOWED BY $pixels PIXELS',
+        labelOffset: markerRect.topCenter + const Offset(0.0, _debugMarkerLabelPadding),
+        rotation: 0.0,
+      ));
+    }
+    if (overflow.bottom > 0.0) {
+      final String pixels = _formatPixels(overflow.bottom);
+      final Rect markerRect = new Rect.fromLTWH(
+        0.0,
+        _overflowContainerRect.height * (1.0 - _kMarkerSize),
+        _overflowContainerRect.width,
+        _overflowContainerRect.height * _kMarkerSize,
+      );
+      regions.add(new _OverflowRegionData(
+        rect: markerRect,
+        label: 'BOTTOM OVERFLOWED BY $pixels PIXELS',
+        labelOffset: markerRect.bottomCenter -
+            const Offset(0.0, _debugMarkerFontSize + _debugMarkerLabelPadding),
+        rotation: 0.0,
+      ));
+    }
+    return regions;
+  }
+
+  void _reportOverflow() {
+    final List<String> overflows = <String>[];
+    if (overflow.left > 0.0)
+      overflows.add(_formatPixels(overflow.left) + ' pixels on the left');
+    if (overflow.top > 0.0)
+      overflows.add(_formatPixels(overflow.top) + ' pixels on the top');
+    if (overflow.bottom > 0.0)
+      overflows.add(_formatPixels(overflow.bottom) + ' pixels on the bottom');
+    if (overflow.right > 0.0)
+      overflows.add(_formatPixels(overflow.right) + ' pixels on the right');
+    String overflowText = '';
+    assert(overflows.isNotEmpty,
+        "Somehow $_containerName didn't actually overflow like it thought it did.");
+    switch (overflows.length) {
+      case 1:
+        overflowText = overflows.first;
+        break;
+      case 2:
+        overflowText = '${overflows.first} and ${overflows.last}';
+        break;
+      default:
+        overflows[overflows.length - 1] = 'and ${overflows[overflows.length - 1]}';
+        overflowText = overflows.join(', ');
+    }
+    FlutterError.reportError(
+      new FlutterErrorDetailsForRendering(
+        exception: 'A $_containerName overflowed by $overflowText.',
```

Comment:
Ah, I see why you wanted `$_containerName`: so that you could still have it say "vertical RenderFlex" or "horizontal RenderFlex". Unfortunately you dropped that feature. :-)